### ISSUE TITLE
Add method to consume bytes without shifting buffer ?

### ISSUE THREAD

**Issue by chifflier** [sozu-proxy/circular#2]
Hi @Geal ,

The `consume` method shifts data if >= half the buffer. I'm using the `circular` crate as a fixed-position buffer (for multi-threading) and need to control when data is moved.

Would you be OK for adding a new method (`consume_noshift` or something ilke that) which just adds `cnt` without shifting data ? I can provide a patch :)

**Comment by Geal** [comment]
Yes, good idea! IIRC @keruspe too encountered an issue with a shift that
happened at the wrong time

**Comment by joeri** [comment]
Unless I'm missing something this was added in 0.3 so this issue can be closed
